### ISSUE TITLE
alignment of pads and double tears possibility

### ISSUE THREAD

**Issue by jano2358** [heikkipu/kicad4devel#9]
hi!

when drawing tracks between two objects of same kind (i.e. two resistors), it may be easy to have them perfectly aligned and the tears have a double ending, if the track is between a resistor and an IC it becomes an impossibility to have the double ending, then drc will comply ending with thousands of errors:

![image](https://user-images.githubusercontent.com/11798801/31074635-dbaaf748-a771-11e7-920e-30b567c8112e.png)

I think this happens when pcbnew adds a small segment to 'complete' the track if they have a difference in alignment, but in the case of the image the added track was of 0.000mm, and to correct this I need to come to legacy canvas, erase the segment and add a teardrop to the last segment:

![image](https://user-images.githubusercontent.com/11798801/31074787-8e24469a-a772-11e7-8c06-b8fd4151b060.png)

this makes the process more tedious.

**Comment by heikkipu** [comment]
Yes, GAL canvas is quite beginning and there are no track specific tools yet. If you break track (legacy canvas again) in two points you can align upper teardrop so that it's shape is rect pad shape as lower pad teardrop. That property works only if track is against pads edge. There are one hidden property, that you can use with rectangular pad shapes. This picture shows what I mean

![image](https://user-images.githubusercontent.com/28824973/31169244-ab7175e4-a900-11e7-8ae1-0b23588b768c.png)

**Comment by jano2358** [comment]
I think we are talking about different things (similar but not the same).
When routing, either on GAL or Legacy, if the centers are not perfectly aligned, the routed adds a small piece of track to complete the route to the center of the pad, then the tear is made in this litle piece of track, like this:
![image](https://user-images.githubusercontent.com/11798801/31170213-964c2614-a8fb-11e7-967b-450bdf4f5645.png)
in this case the missalignement is of 0.00872mm, if I want to have the tear properly done in this case I either need to move the components exactly or delete the small piece of track and add a tear to the next track, this is what makes the process more tedious.
If I disable magnetic pads & tracks, unmark 'limit tracks to 45 degrees', choose free angle mode and allow DRC violations (or several combinations of all this), it happens anyway.

The only posiblilty I can think of to solve this, is to have a bit more of 'tolerance' (maybe up to 0.05mm, something tunable will be the best) to the difference between the possible ending point of the track been drawn and the center (magnetic) point of the pad, then the router will not make the extra track and we have the tears more easily made.

**Comment by jano2358** [comment]
I'm seeing now, if I make the tracks on Legacy and deselect 'limit tracks to 45 degrees' in preferences-> general, I can have the dual ended tears (I don't use Legacy too much). Then the limitation is only on GAL.

**Comment by heikkipu** [comment]
Hi 
Yes, there are very much limitations in GAL with routing and editing. Solving them is quite complicated because, GAL destroy old tracks which are touched in PNS. Example in GAL it is impossible to use different kind of teardrops or arced track corners as in Legacy, If you use and you edit a tracks they are gone and current values are replaced. I have an algo in my mind how to fix it, but it is not as good as in Legacy canvas. WYSIWYG when routing is as hard challenge as too. 
Case like this if you make a node, when routing in GAL or in Legacy too,  somewhere middle of two pads and make sure that tracks angle are stright aligned with pads, which happens in 45 degrees mode and if pads are aligned board in 90 degrees angles,  it will do right kind teardrops in pads. Then there are no small track stubs inside pad. It do right kind teardrops even if rect pads are not aligned 90 degrees, only thing is that track is connected in pad position and is aligned with pad in 90 degrees angle. Track which have teardrop has to be connected in pad position, other way it can be outside of the pad. And if you remove that stub track inside pad and add teardrop, which is junction now, it is not right kind of teardrop.

**Comment by jano2358** [comment]
Hi!
I'm just using it a bit more now, and seems like I didn't check all the combinations before, if (in GAL) I select 'Highlight collisions' and 'Free angle mode' on the router settings, I can have the double ending tears even when the pads are not aligned at all, this solves many situations (the drawback here is the need to be changing the settings all the time). 
Also I'm playing with the properties of the tears himself (width and length) and I'm getting much less errors when check with DRC (100:100:4 is too much for the distances/pad sizes I'm using, 60:40:4 seems more appropriate for 0603/tssop).

**Comment by heikkipu** [comment]
Hi

Playing with values you can get picture of what you can do with these tools. As I mentioned full support is in Legacy canvas at the moment. It is possible to do even these kind of tracks: 

![image](https://user-images.githubusercontent.com/28824973/31851071-2877a2a2-b667-11e7-978e-ff885a778207.png)

**Comment by jano2358** [comment]
I need to practice a bit more to get some of these elongated tears, they seem to have interesting possibilities. Just get today my last design, entirely made with your version of kicad (including the patterns of the left side, those are tracks with the shape copied to the mask layer):
![img_20171027_161628](https://user-images.githubusercontent.com/11798801/32109254-7b30e072-bb34-11e7-8230-e38bbf652b52.jpg)

**Comment by heikkipu** [comment]
Hi.  those elongated teardrops are only possibly to edit in legacy canvas only at the moment. Better UI is needed for them. These two videos show how you can do them. Another video shows howto disable teardrops and rounded corners modes.

https://youtu.be/1t6HCLG-Zg4
https://youtu.be/MM8ODejUDHc

Interesting patterns?
